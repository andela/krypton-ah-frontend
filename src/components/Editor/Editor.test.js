import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { ArticleEditor } from './index';

describe('Editor Component', () => {
  const props = {
    publish: () => {}
  };
  it('should rendered Editor component without crashing', () => {
    const wrapper = shallow(<ArticleEditor {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should change the state after changing the article title', () => {
    const event = { target: { name: 'Learn AI' } };
    const wrapper = shallow(<ArticleEditor />);
    const handleChangeSpy = sinon.spy(wrapper.instance(), 'handleChange');
    wrapper.update();
    wrapper.find('#title').simulate('change', event);
    expect(handleChangeSpy.calledOnce).toBeDefined();
  });
  it('should change the state after changing the article category', () => {
    let event = {};
    event = { target: { selectedIndex: 0 } };
    const index = event.target.selectedIndex;
    event = { target: { options: [index], selectedIndex: 0 } };
    const wrapper = shallow(<ArticleEditor />);
    const handleSelectionSpy = sinon.spy(wrapper.instance(), 'handleSelection');
    wrapper.find('select').simulate('change', event);
    wrapper.update();
    expect(handleSelectionSpy.calledOnce).toBeDefined();
  });
  it('should stimulate on submit event', () => {
    const wrapper = shallow(<ArticleEditor {...props} />);
    wrapper.find('Form').simulate('submit', {
      preventDefault: jest.fn()
    });
    wrapper.update();
    expect(wrapper.find('Form')).toBeDefined();
  });
  it('should save article as draft', () => {
    const wrapper = shallow(<ArticleEditor {...props} />);
    wrapper.find('#draftBtn').simulate('click', {
      preventDefault: jest.fn()
    });
    wrapper.update();
    expect(wrapper.find('#draftBtn')).toBeDefined();
  });

  it('should change the state after changing the article content', () => {
    const event = { target: { name: 'How to learn Nodejs' } };
    const wrapper = shallow(<ArticleEditor />);
    const handleEditorChangeSpy = sinon.spy(wrapper.instance(), 'handleEditorChange');
    wrapper.update();
    wrapper.find('ReactMediumEditor').simulate('change', event);
    expect(handleEditorChangeSpy.calledOnce).toBeDefined();
  });

  it('should handle image upload', () => {
    const event = { target: { files: ['files'] } };
    const wrapper = shallow(<ArticleEditor />);
    const handleEditorChangeSpy = sinon.spy(wrapper.instance(), 'handleImageUpload');
    wrapper.update();
    wrapper.find('#fileUpload').simulate('change', event);
    expect(handleEditorChangeSpy.calledOnce).toBeDefined();
  });
});
