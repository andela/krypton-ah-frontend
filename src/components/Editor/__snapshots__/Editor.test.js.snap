// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor Component should rendered Editor component without crashing 1`] = `
<Form
  as="form"
>
  <FormField
    required={true}
  >
    <label
      htmlFor="title"
    >
      Title
    </label>
    <Input
      className="titleInput"
      id="title"
      name="title"
      onChange={[Function]}
      type="text"
      value=""
    />
  </FormField>
  <FormField
    required={true}
  >
    <label
      htmlFor="description"
    >
      Description
    </label>
    <TextArea
      as="textarea"
      id="description"
      maxLength="250"
      name="description"
      onChange={[Function]}
      rows={3}
      value=""
    />
  </FormField>
  <FormField
    className="categoryForm"
    required={true}
  >
    <label
      className="categoryLabel"
      htmlFor="category"
    >
      Category
    </label>
    <select
      id="category"
      name="category"
      onChange={[Function]}
      placeholder="Select article category"
      value=""
    >
      <option
        disabled={true}
        value=""
      >
        Select Category
      </option>
      <option
        key="1"
        value="Politics"
      >
        Politics
      </option>
      <option
        key="2"
        value="Family"
      >
        Family
      </option>
      <option
        key="3"
        value="Techonology"
      >
        Techonology
      </option>
    </select>
    <hr
      className="categoryLine"
    />
  </FormField>
  <FormField
    required={true}
  >
    <label
      htmlFor="content"
    >
      Content
    </label>
    <ReactMediumEditor
      id="content"
      onChange={[Function]}
      options={
        Object {
          "placeholder": Object {
            "text": "Write your article here...",
          },
          "toolbar": Object {
            "buttons": Array [
              "bold",
              "italic",
              "underline",
              "anchor",
              "strikethrough",
              "subscript",
              "superscript",
            ],
          },
        }
      }
      tag="div"
      text=""
    />
  </FormField>
  <FormField
    required={true}
  >
    <label
      htmlFor="tags"
    >
      Tags
    </label>
    <Input
      id="tags"
      name="tags"
      onChange={[Function]}
      type="text"
      value=""
    />
  </FormField>
  <FormField>
    <label
      htmlFor="featuredImage"
    >
      Featured Image
    </label>
    <div
      className="previewImage"
    >
      <label
        htmlFor="fileUpload"
      >
        <img
          alt="featuredImage"
          className="featuredPreviewImage"
          src="test-file-stub"
        />
      </label>
      <Input
        content="Browse"
        id="fileUpload"
        name="featuredImage"
        onChange={[Function]}
        type="file"
      />
    </div>
  </FormField>
  <div
    className="formButton"
  >
    <Button
      as="button"
      basic={true}
      floated="left"
      id="draftBtn"
      onClick={[Function]}
    >
      <span
        className="publishBtn"
      >
        Save as Draft
      </span>
    </Button>
    <Button
      as="button"
      basic={true}
      floated="right"
      onClick={[Function]}
    >
      <span
        className="publishBtn"
      >
        Publish
      </span>
    </Button>
  </div>
</Form>
`;
